src/comm/Connection.cc

```diff
@@ -180,6 +178,21 @@ Comm::Connection::connectTimeout(const time_t fwdStart) const
     return min(ctimeout, ftimeout);
 }
 
+void
+Comm::Connection::updateIdent(const Ident::Lookup &newLookup)
+{
+    // Multiple updateIdent() calls happen when a configuration directive (that
+    // requires a lookup) is evaluated while a lookup attempt is still pending.
+    // Ident::IdentStateData::notify() makes calls with the same lookup value.
+    static const Ident::User lookupError("[ident-error]");
+    if (identLookup)
```

Comment:
> a4857e9 : Repeated updateIdent() calls for the same Connection are probably relatively rare

These repeated calls are always performed in configurations with configured ident_lookup_access (and ident-using ACLs). The initial ident request is performed via ConnStateData::start()->ConnStateData::whenClientIpKnown().

Reply:
Agreed. That "rare" assertion in commit a4857e93d4148c5139743f57ca47033492ec7864 message was wrong/sloppy -- these repeated calls are frequent in certain configurations, and those configurations may not be so rare if we only look at configurations where Ident is actively used.

Fortunately, the "when they happen..." part is correct/accurate. I still think that code simplification in that commit was the right thing to do overall.